apps/io_tester/io_tester.cc

```diff
@@ -657,6 +665,96 @@ class write_io_class_data : public io_class_data {
     }
 };
 
+class unlink_class_data : public class_data {
+private:
+    sstring _dir_path{};
+    uint64_t _file_id_to_remove{0u};
+
+public:
+    unlink_class_data(job_config cfg) : class_data(std::move(cfg)) {
+        if (!_config.files_count.has_value()) {
+            throw std::runtime_error("request_type::unlink requires specifying 'files_count'");
+        }
+    }
+
+    future<> do_start(sstring path, directory_entry_type type) override {
+        if (type == directory_entry_type::directory) {
+            return do_start_on_directory(path);
+        }
+        throw std::runtime_error(format("Unsupported storage. {} should be directory", path));
+    }
+
+    future<size_t> issue_request(char *buf, io_intent* intent) override {
+        if (all_files_removed()) {
+            return make_ready_future<size_t>(1u);
+        }
+
+        const auto fname = get_filename(_file_id_to_remove);
+        ++_file_id_to_remove;
+
+        return remove_file(fname).then([]{
+            return make_ready_future<size_t>(1u);
+        });
+    }
+
+    void emit_results(YAML::Emitter& out) override {
+        out << YAML::Key << "average latency" << YAML::Value << average_latency();
+    }
+
+private:
+    uint64_t files_count() const {
+        return *_config.files_count;
+    }
+
+    bool all_files_removed() const {
+        return files_count() <= _file_id_to_remove;
+    }
+
+    sstring get_filename(uint64_t file_id) const {
+        return format("{}/test-{}-shard-{:d}-file-{}", _dir_path, name(), this_shard_id(), file_id);
+    }
+
+    future<> do_start_on_directory(sstring path) {
+        _dir_path = std::move(path);
+
+        return seastar::parallel_for_each(boost::irange(UINT64_C(0), files_count()), [this] (uint64_t file_id) {
+            const auto fname = get_filename(file_id);
+            const auto fsize = _config.file_size;
+            const auto flags = open_flags::rw | open_flags::create;
```

Comment:
This is not "unlink" job :) It's create-and-unlink one. I think files should be pre-created before measurement starts

Reply:
Isn't it like that right now? In main() I can see:
```
std::cout << "Creating initial files..." << std::endl;
ctx.invoke_on_all([] (auto& c) {
    return c.start();
}).get();

std::cout << "Starting evaluation..." << std::endl;
ctx.invoke_on_all([] (auto& c) {
    return c.issue_requests();
}).get();
```

Your comment is related to `do_start_on_directory()` function, that is called via `start()`. I thought, that it is completed before the evaluation starts. Does the comment in main() lie? Or maybe I have mistaken something?